packages/core/lib/testing/Version.js

```diff
@@ -0,0 +1,49 @@
+const semver = require("semver");
```

Comment:
What do you think of moving this file to somewhere inside `compile-solidity/compilerSupplier`?

Reply:
It's an interesting idea -- what would it be called?  `versionUtils.js`?

I do feel like it's a little out of place there as these functions are a bit... weird.  Right, like, these "versions" are *sometimes* versions and sometimes ranges.  Do you think that's not a problem?

Reply:
We can use the word "constraint" instead of "version"... or I guess "range" can refer to a range of one?

I like "utils" as part of the name!

Reply:
Well, the thing is that individual versions and version ranges are actually handled slightly separately in order to account for prerelease versions.

It's just confusing because there are three different notions of "version" flying around:

1. Anything that can go in the `version` field (i.e. a version, a range, `native`, or a file path)
2. A version or range
3. A version

IDK -- I guess functions could be renamed to be more explicit?  Instead of `versionUtils` it could be `rangeUtils`?  Does that sound good?

Reply:
Sounds good to me!

Reply:
OK, guess I'll do that then!